File: medium/3121_count_the_number_of_special_characters_ii/solution.py
class Solution:
    def numberOfSpecialChars(self, word: str) -> int:
        lowercase_chars = "abcdefghijklmnopqrstuvwxyz"
        uppercase_chars = "ABCDEFGHIJKLMNOPQRSTUVWXYZ"

        special_chars_count = 0

        for lowercase_c, uppercase_c in zip(lowercase_chars, uppercase_chars):
            lowercase_i = word.rfind(lowercase_c)
            if lowercase_i < 0:
                continue

            uppercase_i = word.find(uppercase_c)
            if uppercase_i < 0:
                continue

            if lowercase_i < uppercase_i:
                special_chars_count += 1

        return special_chars_count

File: medium/3121_count_the_number_of_special_characters_ii/solution.cpp
#include <string>

class Solution {
public:
  int numberOfSpecialChars(std::string word) {
    const std::string lowercaseChars = "abcdefghijklmnopqrstuvwxyz";
    const std::string uppercaseChars = "ABCDEFGHIJKLMNOPQRSTUVWXYZ";

    int specialCharsCount = 0;

    for (size_t i = 0; i < 26; ++i) {
      size_t lowercaseI = word.rfind(lowercaseChars[i]);
      if (lowercaseI == std::string::npos)
        continue;

      size_t uppercaseI = word.find(lowercaseChars[i]);
      if (uppercaseI == std::string::npos)
        continue;

      if (lowercaseI < uppercaseI)
        ++specialCharsCount;
    }

    return specialCharsCount;
  }
};
